src/pages/home/index.js

```diff
@@ -0,0 +1,272 @@
+          cartCounter.classList.add("hidden");
+          cartCounter.innerText = 0;
+        } else {
+          cartCounter.innerText = productsCount;
+        }
+      },
+      { signal: this.abortController.signal },
+    );
+
+    this.components.search.element.addEventListener(
+      "search-filter",
+      (event) => {
+        this.filters.set("_page", "1");
+        this.filters.set("q", event.detail);
+
+        this.updateProductsList();
+      },
+    );
+
+    this.components.sideBar.element.addEventListener(
+      "range-selected",
+      (event) => {
+        const { filterName, value } = event.detail;
+
+        this.filters.set("_page", "1");
+
+        const gte = `${filterName}_gte`;
+        const lte = `${filterName}_lte`;
+
+        this.filters.set(gte, value.from);
+        this.filters.set(lte, value.to);
+
+        this.updateProductsList();
+      },
+    );
+
+    this.components.sideBar.element.addEventListener("add-filter", (event) => {
+      this.filters.set("_page", "1");
+
+      for (const [key, prop] of new URLSearchParams(event.detail)) {
+        this.filters.append(key, prop);
+      }
+
+      this.updateProductsList();
+    });
+
+    this.components.sideBar.element.addEventListener(
+      "remove-filter",
+      (event) => {
+        const [key, prop] = event.detail.split("=");
+        const filters = this.filters
+          .getAll(key)
+          .filter((item) => item !== prop);
+
+        this.filters.set("_page", "1");
+        this.filters.delete(key);
+
+        for (const filter of filters) {
+          this.filters.append(key, filter);
+        }
+
+        this.updateProductsList();
+      },
+    );
+
+    this.components.sideBar.element.addEventListener("clear-filters", () => {
+      this.resetFilters();
+      this.components.search.clear();
+
+      this.updateProductsList();
+    });
+
+    this.components.pagination.element.addEventListener(
+      "page-changed",
+      (event) => {
+        this.filters.set("_page", event.detail + 1);
+
+        this.updateProductsList();
+      },
+    );
+  }
+
+  resetFilters() {
+    this.filters = new URLSearchParams();
+    this.filters.set("_page", "1");
+    this.filters.set("_limit", this.pageLimit);
+  }
+
+  async updateProductsList() {
+    const products = await this.loadProducts();
+
+    this.components.cardsList.update(products);
+  }
+
+  async loadProducts() {
+    const { products, total } = await this.getProducts(this.filters);
+
+    if (total > this.totalPages) {
+      this.totalPages = total;
+    }
+
+    this.components.pagination.update({
+      totalPages: Math.ceil(total / this.pageLimit),
+      page: this.filters.get("_page"),
+    });
+
+    // NOTE: sync backend data with local stored data
+    const productsFromStore = this.productStore.getAll();
+
+    for (const product of products) {
+      if (productsFromStore[product.id]) {
+        product.inStore = true;
+      }
+    }
+
+    return products;
+  }
```

Comment:
**Optimization Issue**: The loadProducts method fetches products based on filters and updates pagination, but it does not handle the scenario where the total number of products is less than the current page, potentially leading to an empty page display without user notification. <br> **Fix**: Add a check after fetching products to see if the current page has products. If not, either adjust the page number to the last available page or show a message indicating no products are available. <br> **Code Suggestion**: 
 ```
 +    if (products.length === 0 && this.filters.get("_page") > 1) {
 +      this.filters.set("_page", Math.ceil(total / this.pageLimit));
 +      this.updateProductsList();
 +    }
 ```